core/services/roleServices.py

```diff
@@ -0,0 +1,8 @@
+from core.models import Role
+
+
+def check_role_unique_name(name, uuid=None):
+    query = Role.objects.filter(name=name, validity_to__isnull=True)
+    if query.exists() and (not uuid or query.get().uuid != uuid):
+        return [{"message": "Location code %s already exists" % name}]
```

Comment:
Update message to replace Location with Role.

Reply:
fixed